### DISCUSSION TITLE
Store sensitive files that include keys on NextJS

### DISCUSSION THREAD

**Discussion by incogiscool** [vercel/next.js#50425] [Help] [answered]
### Summary

Hey, so I am developing a site using NextJS and Firebase. Currently, I am trying to use the admin firebase API  to run a getUsers() function. This requires me to have a JSON file that has all the keys to the API. How can I hide this JSON file so that it's secure and not public?

**Comment by iukea1** [comment]
Possibly look into
.env.local

And also you want to run any code that might have sensitive data (aka might talk to a database and need to auth to it)


You should run the code as a page.ts in the /app directory and that will make it a server component.

No idea if this helps or if I read your question wrong

**Reply by incogiscool** [reply]
Hey, thanks for the response. I am currently still on the /pages directory system, unfortunately. I also have some of the keys set up on .env.local, but for their admin package, they only support it in JSON format for some of the functions (don't ask me why, it's very weird).

I saw something on StackOverflow the other day where people said to store it in the /api folder (the server folder). Would this make it secure?

**Reply by iukea1** [reply, marked as answer]
> Hey, thanks for the response. I am currently still on the /pages directory system, unfortunately. I also have some of the keys set up on .env.local, but for their admin package, they only support it in JSON format for some of the functions (don't ask me why, it's very weird).
> 
> I saw something on StackOverflow the other day where people said to store it in the /api folder (the server folder). Would this make it secure? 

Doing data fetching in react server components  does make things "more secure"

But with a proper /pages directory it has the same level of security.

(My whole background is in security I am learning Full Stack for the sake of the grind 🥸🥸🥸🥸)

**Reply by incogiscool** [reply]
Ahh alright thanks man

ps. Good luck learning full stack haha. I think it should be easy for you since you come from a security background
